core/src/epicli/data/common/ansible/playbooks/roles/rabbitmq/tasks/set-erlang-cookie.yml

```diff
@@ -1,8 +1,43 @@
 ---
-- include_tasks: "cookie-setup/master.yml"
-  when:
-    - groups['rabbitmq'][0] == inventory_hostname
+- name: Set is_cookie_user_defined fact
+  set_fact:
+    is_cookie_user_defined: "{{ specification.cluster.cookie | default('') | length > 0 }}"
 
-- include_tasks: "cookie-setup/nodes.yml"
-  when:
-    - groups['rabbitmq'][0] != inventory_hostname
+- name: Set cookie fact for all hosts if it's defined by user
+  when: is_cookie_user_defined
+  set_fact:
+    erlang_cookie: "{{ specification.cluster.cookie }}"
+  delegate_to: "{{ item }}"
+  delegate_facts: true
+  loop: "{{ ansible_play_hosts }}"
+
+- name: Set cookie fact for all hosts it it's not defined by user
+  when: not is_cookie_user_defined
+  block:
+    - name: Check if .erlang.cookie file exists
+      stat:
+        path: "{{ cookie_file }}"
+      register: cookie_file_stat
+
+    - name: Set cookie fact for all hosts from file if exists
+      when: cookie_file_stat.stat.exists
+      block:
+        - name: Load cookie file data
+          slurp:
+            src: "{{ cookie_file }}"
+          register: cookie_file_data
+
+        - name: Set cookie fact for all hosts from file if exists
+          set_fact:
+            erlang_cookie: "{{ cookie_file_data.content | b64decode | trim }}"
+          delegate_to: "{{ item }}"
+          delegate_facts: true
+          loop: "{{ ansible_play_hosts }}"
+
+    - name: Generate cookie if it's not defined by user and cookie file doesn't exist
+      when: not cookie_file_stat.stat.exists
+      set_fact:
+        erlang_cookie: "{{ lookup('password', '/dev/null length=15 chars=ascii_letters') }}"
```

Comment:
In my opinion that this should be hidden somehow for local log files, @erzetpe ant @to-bar what do you think? 

```
RabbitMQ nodes and the CLI tools use an erlang cookie to determine whether or not they have permission to communicate. The cookie is a string of alphanumeric characters and can be as short or as long as you would like.
```

Reply:
> RabbitMQ nodes and CLI tools (e.g. rabbitmqctl) use a cookie to determine whether they are allowed to communicate with each other. For two nodes to be able to communicate they must have the same shared secret called the Erlang cookie. The cookie is just a string of alphanumeric characters up to 255 characters in size. It is usually stored in a local file. The file must be only accessible to the owner (e.g. have UNIX permissions of 600 or similar). Every cluster node must have the same cookie.

Since Erlang cookie is called "shared secret" and file with it must be only accessible to the owner, I would treat it as sensitive data and use `no_log` parameter here.